### DISCUSSION TITLE
Recap for September 25th, 2024 - Semantic Kernel Office Hours (US/EMEA)

### DISCUSSION THREAD

**Discussion by sophialagerkranspandey** [microsoft/semantic-kernel#9069] [Office Hours Recaps]
Semantic Kernel Office Hours for US/EMEA - September 25th, 2024 recording: https://www.youtube.com/watch?v=QR_DkWv8VoE&list=PLmsFUfdnGr3xAtM3ofjpEkIWnuSA2d-i-&index=34

If you weren't able to attend office hours but still have questions, please post them here: https://github.com/microsoft/semantic-kernel/discussions

**Meeting notes:**

- **Office Hours Overview:** Sophia provided an overview of the office hours format, including the weekly sessions for the US and EMEA, and the monthly sessions for APAC. She also mentioned the new release of version 1.20.0, which includes new features and updates.
- **New Process Framework:** Evan introduced a new process framework launched in the latest .NET build. This framework is designed to help enterprises integrate AI into their business processes, allowing for stateful, reusable, and event-driven steps that can be scaled globally.
- **Process Framework Details:** Evan explained the components of the process framework, including processes, steps, and patterns. He highlighted the flexibility and scalability of the framework, which can handle long-running processes and integrate with existing systems.
- **Deployment Models:** Evan discussed the two deployment models for the process framework: Microsoft Orleans and DAPR. Orleans is suitable for .NET Core, while DAPR supports .NET, Python, and Java, offering more options for state storage.
- 	**Orleans Deployment:** Evan explained that Microsoft Orleans is a deployment model suitable for .NET Core, allowing processes to run on Linux, Windows, and Mac OS. Orleans supports state storage options like Cosmos DB and BLOB storage.
- 	**DAPR Deployment:** Evan described DAPR as another deployment model that supports .NET, Python, and Java. DAPR is open-source and offers more state storage options, making it suitable for different cloud environments and technologies.
- **Step Flexibility:** Evan described the flexibility of steps within the process framework, which can involve code, AI APIs, agents, and human intervention. This allows for dynamic and auditable processes that can adapt to various business needs.
- **Example Process:** Evan provided an example of an account opening process, demonstrating how the process framework can be used to automate and streamline business processes, including steps for credit rating, fraud detection, and customer communication.
- **Coding the Process:** Evan outlined the steps to code a process using the process framework, including creating a new project, defining steps, and setting up patterns. He emphasized the importance of organizing kernel functions, step definitions, and process orchestration for better collaboration and testing.
- **Best Practices:** Evan shared best practices for using the process framework, such as organizing kernel functions, step definitions, and process orchestration into separate components to facilitate collaboration and testing.
- **Function Calling Model:** Sergey announced the release of a new function calling model, which is designed to be reusable across different AI connectors and extensible for future updates. He demonstrated how to set up and use the new model with various examples.
- **Manual Function Invocation:** Sergey explained the manual function invocation mode, which allows users to handle function calls themselves, providing more control over the process. He provided a detailed example of how to implement this mode.
- **Structured Outputs:** Dmytro introduced the structured outputs feature, which allows users to specify a JSON schema for the response format. He demonstrated how to use this feature with examples, highlighting its benefits for strong typing and control over the response format.

**Comment by ksachdeva** [comment]
Dapr has a WorkFlow engine that also builds upon Virtual Actors 
https://docs.dapr.io/developing-applications/building-blocks/workflow/workflow-overview/

any reason why not to use the Workflow building block from Dapr instead of Process?

**Reply by bangonkali** [reply]
@ksachdeva dapr is good as well i believe they are going to support that as well but they will prioritize orleans it seems. See here: https://learn.microsoft.com/en-us/semantic-kernel/frameworks/process/process-deployment, at the bottom

![image](https://github.com/user-attachments/assets/ef7902f2-3213-4b15-a371-10f7732a9514)

**Reply by ksachdeva** [reply]
Actually, they have started with Dapr ... most likely the documentation is out of date
https://github.com/microsoft/semantic-kernel/tree/main/dotnet/src/Experimental/Process.Runtime.Dapr

My question is different. I am wondering why the implementation builds on Dapr Actors instead of directly using Dapr Workflow.

**Reply by bangonkali** [reply]
@ksachdeva thanks for the heads up. They changed the priority it seems. 

I agree with you. Now we will end up with multiple workflow like systems just in dapr alone 😅. I'm also curious why they're not using dapr workflow. Thanks for clarifying your thought on this one.

**Comment by bangonkali** [comment]
Interesting how I am currently working on semantic kernel + orleans and this process framework is introduced to be deployable on orleans. @sophialagerkranspandey is there anywhere we can follow the **Orleans Deployment** story?

**Reply by sophialagerkranspandey** [reply]
Thanks, I'm looping in @alliscode to cover the updates with Orelans

**Reply by bangonkali** [reply]
Hi @alliscode, are there any plans to release examples with orleans?
